snappy/services.go

```diff
@@ -229,62 +190,47 @@ func removePackageServices(s *snap.Info, inter interacter) error {
 	return nil
 }
 
-// ServiceDescription describes a snappy systemd service
-type ServiceDescription struct {
-	SnapName        string
-	AppName         string
-	Version         string
-	Revision        int
-	Description     string
-	SnapPath        string
-	Start           string
-	Stop            string
-	PostStop        string
-	StopTimeout     time.Duration
-	Restart         systemd.RestartCondition
-	Type            string
-	AaProfile       string
-	BusName         string
-	UdevAppName     string
-	Socket          bool
-	SocketFileName  string
-	ListenStream    string
-	SocketMode      string
-	ServiceFileName string
-}
-
-func GenServiceFile(desc *ServiceDescription) string {
+func GenServiceFile(appInfo *snap.AppInfo) string {
 	serviceTemplate := `[Unit]
-Description={{.Description}}
-After=snapd.frameworks.target{{ if .Socket }} {{.SocketFileName}}{{end}}
-Requires=snapd.frameworks.target{{ if .Socket }} {{.SocketFileName}}{{end}}
+Description=service {{.App.Name}} for snap {{.App.Snap.Name}} - autogenerated DO NO EDIT
+After=snapd.frameworks.target{{ if .App.Socket }} {{.SocketFileName}}{{end}}
+Requires=snapd.frameworks.target{{ if .App.Socket }} {{.SocketFileName}}{{end}}
 X-Snappy=yes
 
 [Service]
-ExecStart=/usr/bin/ubuntu-core-launcher {{.UdevAppName}} {{.AaProfile}} {{.FullPathStart}}
+ExecStart=/usr/bin/ubuntu-core-launcher {{.App.SecurityTag}} {{.App.SecurityTag}} {{.FullPathStart}}
 Restart={{.Restart}}
 WorkingDirectory=/var{{.SnapPath}}
```

Comment:
well, then do like for SnapPath for now, but the /var is a hack making assumptions